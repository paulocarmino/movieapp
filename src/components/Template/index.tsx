import Head from 'next/head'
import Logo from '@/components/Logo'

export type TemplateProps = {
  children: JSX.Element
}

const Template = ({ children }: TemplateProps) => {
  return (
    <div className="flex flex-col justify-center items-center px-3 w-full min-h-screen">
      <Head>
        <title>MovieApp</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className='pt-4 w-full xs:w-[700px] h-screen md:w-[768px] lg:w-[1024px]'>
        <div className='mb-4'>
          <Logo />
        </div>

        <div>
          {children}
        </div>
      </main>
    </div>
  )
}

export default Template
