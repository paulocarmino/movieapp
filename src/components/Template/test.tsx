import { fireEvent, render, screen } from '@testing-library/react'

import Template from '.'

describe('<MovieBox />', () => {
  it('should render the component', () => {
    const { container } = render(<Template><div>Oi</div></Template>)

    expect(screen.getByTestId('logo')).toBeInTheDocument()
    expect(container.firstChild).toMatchSnapshot()
  })
})
