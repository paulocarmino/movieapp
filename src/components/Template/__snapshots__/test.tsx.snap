// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MovieBox /> should render the component 1`] = `
<div
  class="flex flex-col justify-center items-center px-3 w-full min-h-screen"
>
  <main
    class="pt-4 w-full xs:w-[700px] h-screen md:w-[768px] lg:w-[1024px]"
  >
    <div
      class="mb-4 cursor-pointer"
      data-testid="logo"
    >
      <div
        class="flex items-center"
      >
        <div>
          <svg
            class="mr-2 text-yellow-400"
            data-testid="logo-icon"
            fill="currentColor"
            height="36"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 512 512"
            width="36"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M371.7 238l-176-107c-15.8-8.8-35.7 2.5-35.7 21v208c0 18.4 19.8 29.8 35.7 21l176-101c16.4-9.1 16.4-32.8 0-42zM504 256C504 119 393 8 256 8S8 119 8 256s111 248 248 248 248-111 248-248zm-448 0c0-110.5 89.5-200 200-200s200 89.5 200 200-89.5 200-200 200S56 366.5 56 256z"
            />
          </svg>
        </div>
        <h1
          class="font-medium text-white"
        >
          MovieApp
        </h1>
      </div>
    </div>
    <div>
      <div>
        Oi
      </div>
    </div>
  </main>
</div>
`;
